QUARKUS-3290.md

```diff
@@ -0,0 +1,35 @@
+# QUARKUS-3290 - Generate source zip which contains the source code of the built artifacts
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-3290
+
+As part of discussions abound QUARKUS-2701, it was decided to generate source zip which contains the source code of the artifact from the internal Gerrit repository for upcoming 3.x release
+
+One of the reasons is to fully comply with https://issues.redhat.com/secure/attachment/12796750/screenshot-1.png from Red Hat compliance training
+
+Another big reason is to demonstrate the amount of work that is done during the productization effort - should help PM in his discussions and with the secure build chain efforts.
+
+## Scope of the testing
+ * Exploratory testing
+ * Evaluation of existing test scenarios
+ * New test development
+
+Topics for exploratory testing:
+ * SRC ZIP naming
+ * Root directories naming
+ * Sources for platform participants
+ * Review of included sources for transitive dependencies
+ * Files count, summary per file extension
+ * Large files check
```

Comment:
Just curious: why check large files?

Reply:
e.g. https://issues.redhat.com/browse/QUARKUS-3293 ;)

Reply:
Maybe we should check against binaries then? Checking first N symbols for non-UTF (non ascii?) content should be trivial enough and give us a reasonable protection.
With size check  would have both false negatives (small binaries) and false positives (huge source code files)